### PR TITLE
allow operator assignment. allow deeper spec folders

### PR DESCRIPTION
**Pull request by AnEmortalKid**
Adds support for some assignment operators to be used not just in loops and statements:

Currently, `foo more 5` => `foo more 5`. 
With this change: `foo more 5` => `foo += 5`. 

Also rewrote some of the test.spec so it can go deeper. Currently, tests are only read at spec/something/something/source.djs. 

Which has lead to dirs like: spec/many/many-is-loop-wow-bar-such-loop-nested.

I think for readability, it helped to support nested loops. Here's some tape output (not piped to tap-dot)

```
# operators/assignment/more/string
ok 18 should be equal
```

Which i find more readable than
```
# rly/rly-but-rly
ok 39 should be equal
```

With this, we could have the following structure:
```
rly/
  source.djs
  expect.js
  but/
   source.djs
   expect.js
    rly/
    source.djs
     expect.js
```

if we wanted to sort of include the nesting in the dirs (we might not want to).

### CODE REVIEW

**Comment by Pholey** [comment]
I actually have some [OS-safe code that does this recursively in deflea](https://github.com/dogescript/deflea/blob/master/scripts/import-specs.js). we could pretty easily adapt it to run tests [as it sort of already does test assertions](https://github.com/dogescript/deflea/blob/master/scripts/import-specs.js#L147)

**Comment by AnEmortalKid** [comment]
https://travis-ci.org/AnEmortalKid/dogescript/builds/289314757

Might have been the walker bit that's not supported in 10 or 11

File: lib/parser.js [hunk 1]
```diff
@@ -186,10 +194,16 @@ module.exports = function parse (line) {
...
-    if (keys[1] === 'is') {
-        statement += keys[0] + ' = ';
-        if (keys[2] === 'new') {
+    // support any kind of assignment operator
+    var assignment = keys[1]
+    if(assignOps.hasOwnProperty(assignment))
+    {
+      statement += keys[0] + assignOps[assignment];
+      
+      // only support new for is, others don't make sense
```
Comment (vpzomtrrfrt) [inline lib/parser.js]: Is there a reason not to allow this?  Disallowing it seems odd and there is a potential that someone will want it to work

File: package.json [hunk 2]
```diff
@@ -15,6 +15,8 @@
   "dependencies": {
     "js-beautify": "^1.6.14",
     "optimist": "^0.6.1",
+    "uglify-js": "^3.1.3",
```
Comment (vpzomtrrfrt) [inline package.json]: Doesn't look like you're using this anywhere, why is it in dependencies?

File: lib/parser.js (hunk 1 above)
Comment (AnEmortalKid) [reply lib/parser.js]: Does it make sense to have: `something += new Something()` ? I didn't think of a use case.

File: package.json (hunk 2 above)
Comment (AnEmortalKid) [reply package.json]: The `min` task uses uglifyjs. I ran `npm run dist` for something and it complained about not having uglify-js`. I guess I can add it as a dev-dependency ( like so: https://github.com/dogescript/dogescript/blob/master/package.json)

File: lib/parser.js (hunk 1 above)
Comment (Pholey) [reply lib/parser.js]: Technically that would compile to something like `something += (new Something).valueOf()` i believe, so we probably want to support this:
`something += new Number(5)`

**Comment by AnEmortalKid** [comment]
@Pholey updated those things a while ago :)

**Pull request merged**